compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -1718,5 +1721,411 @@ class Definitions {
 
       isInitialized = true
     }
+    syntheticSymbolsToComment.foreach { (s, c) =>
+      ctx.docCtx.get.addDocstring(s, Some(c))
+    }
   }
+
+  @tu lazy val syntheticSymbolsToComment: Map[Symbols.Symbol, Comment] = Map(
```

Comment:
Since this map is only used once, I think it'd also be fine to replace it with a def that looks something like:
```scala
   def addSyntheticSymbolsComments(using Context): Unit =
     def add(sym: Symbol, doc: String) =  ctx.docCtx.get.addDocstring(sym, Some(doc))
     add(AnyClass, "/** ... */")
     add(Any_==, "/** ... */")
     ...
```